tasks/Module 5/task_2.py

```diff
@@ -0,0 +1,233 @@
+MOVES = ["north", "south", "west", "east", "l", "r", "u", "d"]
+HERO_SYMBOL = 'H'
+ITEM_SYMBOL = 'X'
+
+
+@dataclass
+class GameItem(Item):
+    """GameItem class."""
+
+    heal: int = 0
+
+    def is_rubbish(self):
+        """Check if Item is rubbish."""
+        return self.name in EXCLUDE_ITEMS
+
+    def is_food(self):
+        """Check if Item is food."""
+        return self.name in FOOD
+
+
+class Hero:
+    """Class Hero."""
+
+    def __init__(self, row, columm):
+        """Init Hero."""
+        super().__init__()
+        self.inv = Inventory()
+        self.stamina = 50
+        self.position = (row - 1, columm - 1)
+        self.can_move = True
+
+    def set_position(self, row, colum):
+        """Set initial posision of hero on a field."""
+        self.position = (row, colum)
+
+    def heal(self, item):
+        """Heal the hero with food."""
+        self.stamina += item.heal
+        print(
+            f"I got the {item.name}. Increased stamina with {item.heal}."
+            f" Current stamia is {self.stamina}"
+        )
+
+    def get_stats(self):
+        """Print hero statistics."""
+        print(f"Stamina {self.stamina}/50\n{self.inv}")
+
+    def drop_item(self, item):
+        """Remove item vrom inventory."""
+        self.inv.remove_item(item)
+
+    def check_heal(self):
+        """Check current stamina value."""
+        if self.stamina <= 0:
+            print(f"You fainted and die, your stamina is {self.stamina}")
+            return -1
+        if self.stamina <= 10:
+            print(f"Your healt is too low - {self.stamina}, you need more food")
+            return 0
+        return 1
+
+    def check_weight(self):
+        """Check weight of backpack and ask for drop."""
+        if not self.can_move:
+            print(
+                f"You are overweighed. Current weight of your backpack is {self.inv.weight}\n"
+                f"Please drop any item from Inventory"
+            )
+            self.get_stats()
+            responce = input("Write an item name to drop: ").rstrip()
+            self.drop_item(responce)
+        if self.inv.check_weight() < 3:
+            self.can_move = True
+        return self.can_move
+
+
+class Board:
+    """Class Board."""
+
+    def __init__(self, columns, rows):
+        """Init Board class."""
+        self.field = [[" " for c in range(columns)] for r in range(rows)]
+        self.columns = columns
+        self.rows = rows
+
+    def set_hero(self, hero):
+        """Set hero to position."""
+        self.field[hero.position[0]][hero.position[1]] = HERO_SYMBOL
+
+    def place_items(self):
+        """Initialize field with items."""
+        for row in self.field:
+            item_name = choice(ITEMS)
+            cost = 0 if item_name in EXCLUDE_ITEMS else randint(1, 5)
+            heal = 0 if item_name not in FOOD else randint(5, 10)
+            random_position = randint(0, len(row) - 1)
+            if row[random_position] != HERO_SYMBOL:
```

Comment:
Two things to consider

- You don't actually need this check, because you are first placing items and then a hero
- If you change the order you would need this, but there is hidden and hard to catch bug here. Think about the case what happens if item is going to be placed on the field occupied by hero. You can get `UnboundLocalError: local variable 'item_position' referenced before assignment` exception on board without hero symbol at the beginning ;)